src/test/llmtemplate_test.cpp

```diff
@@ -0,0 +1,332 @@
+        jinjaFilePath = directoryPath + "/template.jinja";
+    }
+    void TearDown() { py::finalize_interpreter(); }
+
+public:
+    void CreateTokenizerConfig(std::string& fileContents) {
+        CreateConfig(fileContents, tokenizerFilePath);
+    }
+    void CreateJinjaConfig(std::string& fileContents) {
+        CreateConfig(fileContents, jinjaFilePath);
+    }
+};
+
+TEST_F(LLMChatTemplateTest, ChatTemplateEmptyBody) {
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = "";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), false);
+    std::string errorOutput = "Expecting value: line 1 column 1 (char 0)";
+    ASSERT_EQ(finalPrompt, errorOutput);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateEmptyMessage) {
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": []
+        }
+    )";
+    std::string errorOutput = "list object has no element 0";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), false);
+    ASSERT_EQ(finalPrompt, errorOutput);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateSingleMessageError) {
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": {"role": "user", "content": "hello"}
+        }
+    )";
+    std::string errorOutput = "This servable accepts only single message requests";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), false);
+    ASSERT_EQ(finalPrompt, errorOutput);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateDefault) {
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "messages": [{ "content": "How can I help you?" }]
+        }
+    )";
+    std::string expectedOutput = "How can I help you?";
```

Comment:
Just for convenience, can you add comment with standard template so we can deduce what should be the output when reading the tests?